documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,405 @@
+
+   ```
+   $ openssl x509 -req -CA ca-cert -CAkey ca-key -in controller01.csr -out controller01.pem \
+        -days 3650 -CAcreateserial -passin pass:changeit
+   ```
+
+4. Prepare a keystore containing the signed server certificate and the CA's certificate.
+
+   ```
+   # Import the CA certificate into a new keystore file.
+   $ keytool -keystore controller01.server.jks -alias CARoot -noprompt \
+           -import -file ca-cert -storepass changeit
+
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore controller01.server.jks -alias controller01 -noprompt \
+           -import -file controller01.pem -storepass changeit
+   ```
+
+5. Export each server's key into a separate `PEM` file.
+
+   This is a two step process.
+   * First, convert the server's keystore in `.jks` format into `.p12` format.
+
+     ```
+     keytool -importkeystore -srckeystore controller01.jks  \
+                           -destkeystore controller01.p12 \
+                           -srcstoretype jks -deststoretype pkcs12 \
+                           -srcstorepass changeit -deststorepass changeit
+    ```
+   * Then, export the private key of the server into a `PEM` file. Note that the generated `PEM` file is not protected
+   by a password. The key itself is password-protected, as we are using the `-nodes` flag. So, be sure
+   to protect it using operating system's technical controls as well as procedural controls.
+
+     ```
+     openssl pkcs12 -in controller01.p12 -out controller01.key.pem -passin pass:1111_aaaa -nodes
+     ```
+
+Step 5 concludes this stage, and the stage is now set for installing the certificates and other PKI material in Pravega.
+
+The table below lists the key output of this stage. Note that you'll typically need one of each file per Pravega
+service, but you may share the same file for services collocated on the same host for logistical or economical reasons.
+
+| Files | Example File| Description |
+|:-----:| :---:|:--------|
+| Certificate in `PEM` format | `controller01.pem` file | The signed certificate to be used by the service.|
+| Private key in `PEM` format | `controller01.key.pem` file  |  The private key to be used by the service. |
+| Server keystore in `JKS` format | `controller01.server.jks` file | The keystore file to be used by the service. |
+
+
+### Deploying certificates and enabling TLS in Pravega
+
+We'll discuss this in the [next](#enabling-tls-and-auth-in-pravega) section, together with other security configuration and setup.
+
+## Enabling TLS and Auth in Pravega
+
+Enabling TLS and auth in Pravega involves the following steps:
+
+1. Configuring TLS and auth Parameters for the services
+2. Configuring TLS and credentials on the client Side
+3. Having the TLS and auth parameters take effect
+
+Each of the above tasks are discussed in the following sub-sections.
+
+### Configuring TLS and Auth Parameters for for the Services
```

Comment:
remove for (repeated)

Reply:
Thanks for catching it! :) Have fixed it in the latest commit.